pkg/imageset/generate_test.go

```diff
@@ -35,8 +35,12 @@ func TestGenerateImageSetJob(t *testing.T) {
 
 func testClusterDeployment() *hivev1.ClusterDeployment {
 	cd := &hivev1.ClusterDeployment{}
+	cd.Spec.PullSecret = &corev1.LocalObjectReference{
+		Name: testPullSecretName,
+	}
```

Comment:
Per call lets remove this and fix where the failure was, it will need to be updated to use the new cd.Status.PullSecret field, if set, and if that is not set, it should skip adding the volume entirely.

Reply:
Right, this is very good catch.